tests/test_coordinates/test_diff_transform.py

```diff
@@ -224,6 +224,26 @@ def test_velocity2(spherical_differential, bl_differential):
     assert_allclose(bd.velocity(metric=mk)[1:], [bd.v_r.value, bd.v_th.value, bd.v_p.value])
 
 
+def test_Coordinateerror_class():
+    """
+    Tests, if the errors raised with the CoordinateError class \
+    ``einsteinpy.coordinates.utils.CoordinateError`` are \
+    appropriate
+
+    """
+    err = CoordinateError(
+        "This type of error is raised "
+        "when there is a coordinate system mismatch"
+        "in metric object"
+        )
+
+    assert type(str(err)) == str
+    assert type(repr(err)) == str
+
```

Comment:
@JeS24  Hey! I'm not sure how to proceed here. I've done basic  type checking for `__str__` and `__repr__`. How should I check for the content of the string ?

Reply:
You do not need to test, whether the exception is being raised, as it is already being tested at appropriate places. All you need to test, for a `BaseError` or `CoordinateError` object (say `err`), are as follows to get 100% coverage:
1. Whether the constructor takes in parameters of different data-types without any error.
2. Whether `str(err) == repr(err)`.
3. Whether `hasattr(err, "message") is True`.
4. Whether `err.message == <whatever values the object was initialized with>`. (Make sure to use a mix of types, e.g., `err = CoordinateError("Error", 404, chr(0xa), "\bPage Found =", False, "\b, Please contact :", 11.223344)`)

Moreover, it's better to move tests for exceptions over to `einsteinpy/tests/test_utils/test_exceptions.py`.